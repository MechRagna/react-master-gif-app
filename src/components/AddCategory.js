import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length > 3) {
      setCategories((cats) => {
        if (!cats.includes(inputValue.toLowerCase())) {
          return [inputValue.toLowerCase(), ...cats];
        } else {
          return cats;
        }
      });
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
