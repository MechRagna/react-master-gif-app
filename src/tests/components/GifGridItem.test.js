import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";
import "@testing-library/jest-dom";

describe("Validaciones de GifGridItem", () => {
  const title = "Titulo de prueba";
  const url = "http://localhost/test.gif";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Validar que se vea bien el componente ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Validar el parrafo", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Validar la URL de la img", () => {
    const img = wrapper.find("img");
    expect(img.props().src).toBe(url);
  });

  test("Validar el ALT de la img", () => {
    const img = wrapper.find("img");
    expect(img.props().alt).toBe(title);
  });
  
  // animate__backInUp
  test("Validar las class del div", () => {
    const div = wrapper.find("div");
    // console.log("div.hasClass(): ", div.hasClass("animate__backInUp"));
    expect(div.hasClass("animate__backInUp")).toBe(true);
  });

});
