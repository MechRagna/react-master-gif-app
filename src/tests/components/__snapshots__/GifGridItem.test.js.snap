// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validaciones de GifGridItem Validar que se vea bien el componente  1`] = `
<div
  className="card animate__animated animate__backInUp"
>
  <img
    alt="Titulo de prueba"
    src="http://localhost/test.gif"
  />
  <p>
    Titulo de prueba
  </p>
</div>
`;
